src/otx/algorithms/common/utils/__init__.py

```diff
@@ -28,6 +28,7 @@
     UncopiableDefaultDict,
     cast_bf16_to_fp32,
     get_arg_spec,
+    get_cfg_based_on_device,
```

Comment:
Will that implicit replacement of configs affect model accuracy?

Reply:
It can be, but as you can see in experiment results, it doesn't decrease model score compared to original setting.
And also it's only applied only if user uses XPU and YOLOX variants models or instance segmentation models. Therefore, it's effect is restrictive, I think.

Reply:
Initially, I was trying to preserve the same training strategy for all devices, and still I don't think this should be device specific. If we change the configs to avoid dynamic shapes, the changes should be applied to all devices. Otherwise, we can't compare CUDA vs XPU accuracy & e2e time directly. 
The experiments from this PR show that the new strategy can replace the original one for the CUDA case . Perhaps, additional discussion is required